Travel Scheduler/Views/PlaceView.m

```diff
@@ -64,79 +64,88 @@ void reformatOverlaps(UILabel *name, UILabel *times, CGRect cellFrame)
     }
 }
 
+UIImageView *instantiateLockImageView(UILabel *lateralLabel)
+{
+    int sideSize = lateralLabel.frame.size.height;
+    int xCoord = lateralLabel.frame.origin.x + lateralLabel.frame.size.width + 10;
+    int yCoord = lateralLabel.frame.origin.y;
+    UIImageView *imageView = [[UIImageView alloc] initWithFrame:CGRectMake(xCoord, yCoord, sideSize, sideSize)];
+    UIImage *lockImage = [UIImage imageNamed:@"lockIcon"];
+    imageView.image = lockImage;
+    imageView.contentMode = UIViewContentModeScaleAspectFill;
+    imageView.clipsToBounds = YES;
+    return imageView;
+}
+
 @implementation PlaceView
 
 #pragma mark - PlaceView lifecycle
 
 - (instancetype)initWithFrame:(CGRect)frame andPlace:(Place *)place
 {
     self = [super initWithFrame:frame];
-    self.color = (place.scheduledTimeBlock % 2 == 0) ? [UIColor orangeColor] : [UIColor blueColor];
-    if (place.locked) {
-        self.layer.borderWidth = 2;
-        [self.layer setBorderColor: [[UIColor redColor] CGColor]];
+    CAGradientLayer *gradient = [CAGradientLayer layer];
+    gradient.frame = self.bounds;
+    if(place.scheduledTimeBlock % 2 == 0) {
+        gradient.colors = @[(id)getColorFromIndex(ExodusFruit).CGColor, (id)getColorFromIndex(ShyMoment).CGColor];
+        [self.layer insertSublayer:gradient atIndex:0];
+    } else {
+        gradient.colors = @[(id)getColorFromIndex(RegularPink).CGColor, (id)getColorFromIndex(LightPink).CGColor];
     }
-    self.backgroundColor = [self.color colorWithAlphaComponent:0.25];
+    [self.layer insertSublayer:gradient atIndex:0];
+    
+    self.layer.shadowOffset = CGSizeMake(1, 0);
+    self.layer.shadowColor = [[UIColor blackColor] CGColor];
+    self.layer.shadowRadius = 5;
+    self.layer.shadowOpacity = .25;
+    self.clipsToBounds = false;
+    self.layer.masksToBounds = false;
+    
+    
+    self.backgroundColor = [self.color colorWithAlphaComponent:0.9];
     _place = place;
-    if (45 < CGRectGetHeight(self.frame) - 10) {
-        self.placeImage = makeImage(self.place.iconUrl);
-        [self addSubview:self.placeImage];
-    }
     [self makeLabels];
     [self makeEditButton];
-    [self createGestureRecognizers];
```

Comment:
Wait, did you delete the gesturerecognizers?

Reply:
oh no! I only put them in another place inside the same method :)